frontend/src/components/dashboard/customHomePage/CustomHomePage.js

```diff
@@ -51,14 +98,104 @@ export class CustomHomePage extends Component {
     });
   };
 
+  handleVariancePageChange = (page, per_page) => {
+    this.setState({ variancesLoaded: false });
+    const params = { ...this.state.params, page, per_page };
+    return this.props.fetchVariances(params).then(() => {
+      this.setState({ variancesLoaded: true, params });
+    });
+  };
+
+  handleUpdateVariance = (files, variance, isApproved) => (values) => {
+    // if the application isApproved, set issue_date to today and set expiry_date 5 years from today,
+    // unless the user sets a custom expiry.
+    const { variance_document_category_code } = values;
+    const issue_date = isApproved ? moment().format("YYYY-MM-DD") : null;
+    let expiry_date;
+    if (isApproved) {
+      expiry_date = values.expiry_date
+        ? values.expiry_date
+        : moment(issue_date, "YYYY-MM-DD").add(5, "years");
+    }
+    const newValues = { ...values, issue_date, expiry_date };
+    const mineGuid = variance.mine_guid;
+    const varianceGuid = variance.variance_guid;
+    const codeLabel = this.props.complianceCodesHash[variance.compliance_article_id];
+    this.props.updateVariance({ mineGuid, varianceGuid, codeLabel }, newValues).then(async () => {
+      await Promise.all(
+        Object.entries(files).map(([document_manager_guid, document_name]) =>
+          this.props.addDocumentToVariance(
+            { mineGuid, varianceGuid },
+            {
+              document_manager_guid,
+              document_name,
+              variance_document_category_code,
+            }
+          )
+        )
+      );
+      this.props.closeModal();
+      this.props.fetchVariances(this.state.params).then(() => {
+        this.setState({ variancesLoaded: true });
+      });
+    });
+  };
+
+  openEditVarianceModal = (variance) => {
+    this.props.openModal({
+      props: {
+        onSubmit: this.handleUpdateVariance,
+        title: this.props.complianceCodesHash[variance.compliance_article_id],
+        mineGuid: variance.mine_guid,
+        mineName: variance.mine_name,
+        varianceGuid: variance.variance_guid,
+      },
+      content: modalConfig.EDIT_VARIANCE,
+    });
+  };
+
+  openViewVarianceModal = (variance) => {
+    this.props.openModal({
+      props: {
+        variance,
+        title: this.props.complianceCodesHash[variance.compliance_article_id],
+        mineName: variance.mine_name,
+      },
+      content: modalConfig.VIEW_VARIANCE,
+      isViewOnly: true,
+    });
+  };
+
+  handleFilterChange = (pagination, filters) => {
+    const { status } = filters;
+    this.setState({ variancesLoaded: false });
+    const params = { ...this.state.params, variance_application_status_code: [...status], page: 1 };
```

Comment:
`[...status]` is this spreading an array inside of an array? If so, that's a noop. You can just use `status` instead

Reply:
removed!